apps/studio/data/config/project-compliance-config-mutation.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { toast } from 'sonner'
+
+import type { ResponseError } from 'types'
+import { configKeys } from './keys'
+import { handleError, patch } from 'data/fetchers'
+
+export type ComplianceConfigUpdateVariables = {
+  projectRef: string
+  isSensitive: boolean
+}
+
+export type ComplianceConfigUpdateResponse = {
+  is_sensitive: boolean
+  error?: any
+}
+
+export async function updateComplianceConfig({
+  projectRef,
+  isSensitive,
+}: ComplianceConfigUpdateVariables) {
+  if (!projectRef) throw new Error('projectRef is required')
+
+  const { data, error } = await patch('/platform/projects/{ref}/settings/sensitivity', {
+    params: {
+      path: { ref: projectRef },
+    },
+    body: { is_sensitive: isSensitive },
+  })
+
+  if (error) handleError(error)
+  return data
+}
+
+type ComplianceConfigUpdateData = Awaited<ReturnType<typeof updateComplianceConfig>>
+
+export const useComplianceConfigUpdateMutation = ({
+  onSuccess,
+  onError,
+  ...options
+}: Omit<
+  UseMutationOptions<ComplianceConfigUpdateData, ResponseError, ComplianceConfigUpdateVariables>,
+  'mutationFn'
+> = {}) => {
+  const queryClient = useQueryClient()
+
+  return useMutation<ComplianceConfigUpdateData, ResponseError, ComplianceConfigUpdateVariables>(
+    (vars) => updateComplianceConfig(vars),
+    {
+      async onSuccess(data, variables, context) {
+        const { projectRef } = variables
+        await queryClient.invalidateQueries(configKeys.projectComplianceConfig(projectRef))
```

Comment:
noticed that the is_sensitive data is from the project settings V2 query, the invalidation should invalidate that query instead then so that the UI is updated